graph/src/components/metrics/stopwatch.rs

```diff
@@ -0,0 +1,156 @@
+use crate::prelude::*;
+use std::collections::HashMap;
+use std::sync::{atomic::AtomicBool, atomic::Ordering, Mutex};
+use std::time::Instant;
+
+/// This is a "section guard", that closes the section on drop.
+pub struct Section {
+    id: String,
+    stopwatch: StopwatchMetrics,
+}
+
+impl Section {
+    /// A more readable `drop`.
+    pub fn end(self) {}
+}
+
+impl Drop for Section {
+    fn drop(&mut self) {
+        self.stopwatch
+            .end_section(std::mem::replace(&mut self.id, String::new()))
+    }
+}
+
+#[derive(Clone)]
+pub struct StopwatchMetrics {
+    disabled: Arc<AtomicBool>,
+    inner: Arc<Mutex<StopwatchInner>>,
+}
+
+impl StopwatchMetrics {
```

Comment:
Could you add a comment on the typical use of this, i.e., that you do `let _section = start_section(..)` and then have `drop` end the section? It required me to look into the implementation to understand how the uses in the rest of the PR worked.

Reply:
I tried doing that, let me know if it's helpful.